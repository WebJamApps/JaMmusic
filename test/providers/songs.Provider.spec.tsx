/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import superagent from 'superagent';
import renderer, { act } from 'react-test-renderer';
import SProvider from '../../src/providers/Songs.provider';
import testSongs from '../testSongs';

describe('the songs provider', () => {
  it('fetches the songs', async () => {
    const res:any = { body: testSongs };
    const sa:any = { set: () => Promise.resolve(res) };
    superagent.get = jest.fn(() => sa);
    let component: renderer.ReactTestRenderer;
    await act(async () => {
      component = renderer.create(<SProvider><div /></SProvider>);
      if (component)expect(component.toJSON()).toMatchSnapshot();
    });
    expect(superagent.get).toBeCalledWith(`${process.env.BackendUrl}/song`);
  });
  it('catches error when fetches the songs', async () => {
    const sa:any = { set: () => Promise.reject(new Error('bad')) };
    superagent.get = jest.fn(() => sa);
    await act(async () => {
      renderer.create(<SProvider><div /></SProvider>);
    });
    expect(superagent.get).toBeCalledWith(`${process.env.BackendUrl}/song`);
  });
});
