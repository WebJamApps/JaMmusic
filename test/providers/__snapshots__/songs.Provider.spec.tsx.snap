// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the songs provider fetches the songs 1`] = `null`;
